src/bootstrap/src/utils/helpers.rs

```diff
@@ -549,7 +549,12 @@ pub fn hex_encode<T>(input: T) -> String
 where
     T: AsRef<[u8]>,
 {
-    input.as_ref().iter().map(|x| format!("{:02x}", x)).collect()
+    use std::fmt::Write;
+
+    input.as_ref().iter().fold(String::new(), |mut acc, &byte| {
```

Comment:
Could we pre-allocate the string here?

Reply:
Good idea, thanks! (https://github.com/rust-lang/rust/compare/910eaae9048b243a247cfbf472d945a0c85757e8..c5fdb391208b309959a63f9798f85b0179bf947e).